crates/ruma-events/src/sticker.rs

```diff
@@ -26,11 +31,142 @@ pub struct StickerEventContent {
 
     /// The URL to the sticker image.
     pub url: OwnedMxcUri,
+
+    /// The media source
+    #[cfg(not(feature = "compat-encrypted-stickers"))]
+    #[serde(skip)]
+    pub source: MediaSource,
 }
 
 impl StickerEventContent {
     /// Creates a new `StickerEventContent` with the given body, image info and URL.
+    #[cfg(not(feature = "compat-encrypted-stickers"))]
+    pub fn new(body: String, info: ImageInfo, url: OwnedMxcUri) -> Self {
+        Self { body, info, url: url.clone(), source: MediaSource::Plain(url.clone()) }
+    }
+    /// Creates a new `StickerEventContent` with the given body, image info and URL.
+    #[cfg(feature = "compat-encrypted-stickers")]
     pub fn new(body: String, info: ImageInfo, url: OwnedMxcUri) -> Self {
         Self { body, info, url }
     }
+
+    /// Creates a new `StickerEventContent` with the given body, image info, URL, and media source.
+    #[cfg(not(feature = "compat-encrypted-stickers"))]
+    pub fn from_source(
+        body: String,
+        info: ImageInfo,
+        url: OwnedMxcUri,
+        source: MediaSource,
+    ) -> Self {
+        Self { body, info, url, source }
+    }
+    /// Creates a new `StickerEventContent` with the given body, image info and URL.
+    #[cfg(feature = "compat-encrypted-stickers")]
+    pub fn from_source(
+        body: String,
+        info: ImageInfo,
+        url: OwnedMxcUri,
+        _source: MediaSource,
+    ) -> Self {
+        Self { body, info, url }
+    }
+}
+
+impl<'de> Deserialize<'de> for StickerEventContent {
```

Comment:
You shouldn't need to implement `Deserialize` for `StickerEventContent`, we should rely on the deserialization of the new `MediaSource`, like we do on `ImageMessageEventContent`.